### PR TITLE
Toolbar Styles

### PR DESCRIPTION
**Pull request by petemill**
- 😎 Introduces Private Window theme (purple)
- 🎨 Refines dark theme (deprioritized light theme for now since there are no releases which default to it)
- 📐 Shapes for Tabs, LocationBar and Buttons

Fix https://github.com/brave/brave-browser/issues/670 (needs follow-up for light theme at https://github.com/brave/brave-browser/issues/961)
Fix https://github.com/brave/brave-browser/issues/785
Fix https://github.com/brave/brave-browser/issues/787 (needs follow-up for bottom corner via https://github.com/brave/brave-browser/issues/962)
Fix https://github.com/brave/brave-browser/issues/786

image

image

image

## Architecture:
- Introduces `layout_constants` override
- Introduces `omnibox_theme` override
- Theme Service now centralises getting the configured theme and the active theme (i.e. working out which theme is active based on a configured default value and the build channel - dark for dev channel)

## Patching
Many things are tied to the static function `LocationBarView::IsRounded()`. Unfortunately, we want this to return `false` in some circumstances (e.g. drawing the border and the background), and not in others (e.g. drawing the omnibox popup).
Therefore, this PR adds a new static function: `LocationBarView::ShouldOmniboxSurround()`.

We can't use a chromium_src override for omnibox_popup_contents_view.cc to define `IsRounded` as `ShouldOmniboxSurround`, since this includes the LocationBarView header file. We also need some, but not all calls from LocationBarView members to change which static function is called, so patching is the only method I've found so far...

## TODO
- [x] Nicer patching to get a different border radius on LocationBar
- [ ] Tests - https://github.com/brave/brave-browser/issues/963
- [ ] ~More color and shape refinements~ <-- Subsequent PR

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Needed or QA/No-QA-Needed) to include the closed issue in milestone 

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by petemill** [comment]
@bbondy @simonhong This is ready to begin review. I've updated the description and created follow-up issues, but I think this is a nice start. I'm doing a final cross-platform manual test now too.

File: browser/themes/brave_theme_service.cc [hunk 1]
```diff
@@ -6,15 +6,18 @@
 
 #include <string>
 
+
```
Comment (simonhong) [inline browser/themes/brave_theme_service.cc]: nit: new line

File: browser/themes/brave_theme_service.cc [hunk 2]
```diff
@@ -56,12 +85,16 @@ void BraveThemeService::Init(Profile* profile) {
   ThemeService::Init(profile);
 }
 
+
+
```
Comment (simonhong) [inline browser/themes/brave_theme_service.cc]: ditto

File: chromium_src/chrome/browser/ui/omnibox/omnibox_theme.cc [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    case OmniboxTint::PRIVATE:
+      return priv;
+    default:
+      return light;
+  }
+}
+
+}
+
+const SkColor kPrivateLocationBarBackground = SkColorSetRGB(0x1b, 0x0e, 0x2c);
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/omnibox/omnibox_theme.cc]: How about moving this constant to anonymous namespace?

File: chromium_src/chrome/browser/ui/omnibox/omnibox_theme.cc [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      return OmniboxTint_Chromium::LIGHT;
+    case OmniboxTint::NATIVE:
+      return OmniboxTint_Chromium::NATIVE;
+    default:
+      return OmniboxTint_Chromium::LIGHT;
+  }
+}
+
+template <class T>
+constexpr T DarkPrivateLight(OmniboxTint tint,
```
Comment (simonhong) [inline chromium_src/chrome/browser/ui/omnibox/omnibox_theme.cc]: This template is needed? I found that only SkColor is used as template argument.

**Comment by petemill** [comment]
All feedback is now addressed - thanks @simonhong

**Comment by bbondy** [comment]
0.55.x 2d326149ea49d8c59254bca7bc656a8fd19f5b75

**Pull request merged**